h2o-docs/src/product/welcome.rst

```diff
@@ -893,3 +893,118 @@ After obtaining the IP address, point your browser to the specified ip address a
+
+Creating the Headless Service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, a headless service must be created on Kubernetes:
+
+::
+
+  apiVersion: v1
+  kind: Service
+  metadata:
+    name: h2o-service
+  spec:
+    type: ClusterIP
+    clusterIP: None
+    selector:
+      app: h2o-k8s
+    ports:
+    - protocol: TCP
+      port: 54321
+
+The ``clusterIP: None`` defines the service as headless. ``The port: 54321`` is the default H2O port. Users and client libraries use this port to talk to the H2O cluster.
+
+The ``app: h2o-k8s`` setting is of **great importance** because it is the name of the application with H2O pods inside. While the name is arbitrarily chosen for this example, it **must** correspond to the chosen H2O deployment name. 
+
+Creating the H2O Deployment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is strongly recommended to run H2O as a `Stateful Set <https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/>`__ on a Kubernetes cluster. Treating H2O nodes as stateful ensures that:
+
+- H2O Nodes are treated as a single unit. They will be brought up and down gracefully and together.
+- No attempts will be made by a K8S healthcheck to restart individual H2O Node in case of an error.
+- The cluster will be restared as a whole (if required).
+- Persistent storages and volumes associated with the Stateful Set of H2O Nodes will not be deleted once the cluster is brought down.
+
+::
+
+  apiVersion: apps/v1
+  kind: StatefulSet
+  metadata:
+    name: h2o-stateful-set
+    namespace: h2o-statefulset
+  spec:
+    serviceName: h2o-service
+    replicas: 3
+    selector:
+      matchLabels:
+        app: h2o-k8s
+    template:
+      metadata:
+        labels:
+          app: h2o-k8s
+      spec:
+        terminationGracePeriodSeconds: 10
+        containers:
+          - name: h2o-k8s
+            image: '<someDockerImageWithH2OInside>'
+            resources:
+              requests:
+                memory: "4Gi"
+            ports:
+              - containerPort: 54321
+                protocol: TCP
+            env:
+            - name: H2O_KUBERNETES_SERVICE_DNS
+              value: h2o-service.h2o-statefulset.svc.cluster.local
+            - name: H2O_NODE_LOOKUP_TIMEOUT
+              value: '180'
+            - name: H2O_NODE_EXPECTED_COUNT
+              value: '3'
+
+There are several other environment variables:
+
+- ``H2O_KUBERNETES_SERVICE_DNS`` - **[MANDATORY]** Crucial for the clustering to work: the format usually follows the ``<service-name>.<project-name>.svc.cluster.local`` pattern. This setting enables H2O node discovery via DNS. It must be modified to match the name of the headless service created.
+- ``H2O_NODE_LOOKUP_TIMEOUT`` - **[OPTIONAL]** Node lookup constraint: time before the node lookup is ended (defaults to 3 minutes).
```

Comment:
- ``H2O_NODE_LOOKUP_TIMEOUT`` (Optional): The node lookup constraint. Specify the time before the node lookup times out. (Defaults to 3 minutes.)